crates/walrus-service/src/client/daemon/routes.rs

```diff
@@ -210,33 +299,42 @@ pub(super) async fn get_blob<T: WalrusReadClient>(
         Ok(consistency_check) => consistency_check,
         Err(error) => return error.into_response(),
     };
-    match client.read_blob(&blob_id, consistency_check).await {
-        Ok(blob) => {
-            tracing::debug!("successfully retrieved blob");
-            let mut response = (StatusCode::OK, blob).into_response();
-            let headers = response.headers_mut();
-            populate_response_headers_from_request(
-                request_method,
-                &request_headers,
-                &blob_id.to_string(),
-                headers,
-            );
-            response
-        }
-        Err(error) => {
-            let error = GetBlobError::from(error);
 
-            match &error {
-                GetBlobError::BlobNotFound => {
-                    tracing::debug!(?blob_id, "the requested blob ID does not exist")
-                }
-                GetBlobError::Internal(error) => tracing::error!(?error, "error retrieving blob"),
-                _ => (),
+    let mut response = if let Some(range_header) = request_headers.get(RANGE) {
```

Comment:
Might need to check that `axum` is not already handling some sort of slicing here. I'm not sure.

Reply:
Sounds good!